src/main/java/org/sonarsource/scanner/ant/SonarQubeTask.java

```diff
@@ -48,13 +57,37 @@ public void execute() {
 
   // VisibleForTesting
   void launchAnalysis(Properties properties) {
-    EmbeddedRunner.create()
-      .addProperties(properties)
+    EmbeddedRunner runner = EmbeddedRunner.create(new LogOutputImplementation())
+      .addGlobalProperties(properties)
       .unmask("org.apache.tools.ant")
       .unmask("org.sonar.ant")
-      .setApp("Ant", SonarQubeTaskUtils.getTaskVersion())
-      .addExtensions(getProject())
-      .execute();
+      .setApp("Ant", SonarQubeTaskUtils.getTaskVersion());
+
+    runner.start();
+    try {
+      runner.addExtensions(getProject());
+    } catch (Exception e) {
+      // Not supported in recent SQ versions. Ignore
+    }
+    runner.runAnalysis(properties);
+    runner.stop();
+
   }
 
+  private static int toAntLevel(Level level) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "toAntLevel" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)